### PR TITLE
Add tests to increase code coverage for 'logic' components

### PR DESCRIPTION
**Pull request by ianyong**
Changes:
- Bring code coverage for the 'logic' components up to 100%.

Related to #115.

### CODE REVIEW

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/DeleteExpenseCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+        assertCommandSuccess(deleteExpenseCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexFilteredList_throwsCommandException() {
+        showExpenseAtIndex(model, INDEX_FIRST_TRANSACTION);
+
+        Index outOfBoundIndex = INDEX_SECOND_TRANSACTION;
+        // ensures that outOfBoundIndex is still in bounds of finance tracker list
```
Comment (yongping827) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/DeleteExpenseCommandTest.java:77]:
```suggestion
        // ensures that outOfBoundIndex is still within bounds of finance tracker list of expenses
```

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/DeleteIncomeCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+        assertCommandSuccess(deleteIncomeCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexFilteredList_throwsCommandException() {
+        showIncomeAtIndex(model, INDEX_FIRST_TRANSACTION);
+
+        Index outOfBoundIndex = INDEX_SECOND_TRANSACTION;
+        // ensures that outOfBoundIndex is still in bounds of finance tracker list
```
Comment (yongping827) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/DeleteIncomeCommandTest.java:77]:
```suggestion
        // ensures that outOfBoundIndex is still within bounds of finance tracker list of incomes
```

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/DeleteCommandTest.java [hunk 3]
```diff
@@ -75,7 +75,7 @@ public void execute_invalidIndexFilteredList_throwsCommandException() {
```
Comment (yongping827) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/DeleteCommandTest.java:75]: Line 73
```
// ensures that outOfBoundIndex is still within bounds of finance tracker list of transactions
```

**Review by yongping827 — CHANGES_REQUESTED** [review]

**Review by yongping827 — APPROVED** [review]
LGTM

**Review by zhaojj2209 — APPROVED** [review]
LGTM!

**Pull request merged**